nodeJS/APIs/APIs.md

```diff
@@ -12,41 +12,42 @@ By the end of this lesson, you should be able to do the following:
 
-- To get a list of blog posts you would `GET` `/posts`, so in your express app you would respond to  `app.get("/posts")` with a list of all posts.
-- To add a new blog-post you would `POST` to `/posts`.
-- To update the contents of a blog-post you would `PUT` to the URI of the specific post you're updating. `/posts/:postid`
-- To delete a post you would `DELETE` `/posts/:postid`
+Each part of an API URI specifies the resource. For example, `GET /posts` would return the entire list of blog posts while `GET /posts/:postid` specifies the exact blog post we want. We could nest further with `GET /posts/:postid/comments` to return a list of comments for that blog post or even `GET /posts/:id/comments/:commentid` for a very specific blog post comment.
 
 ### CORS
 The [Same Origin Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) is an important security measure that basically says "Only requests from the same origin (the same IP address or URL) should be allowed to access this API". (Look at the link above for a couple of examples of what counts as the 'same origin'.) This is a big problem for us because we are specifically trying to set up our API so that we can access it from different origins, so to enable that we need to set up Cross-origin resource sharing, or CORS.
 
-Setting up CORS in express is very easy, there's a middleware that does the work for us. [This article](https://medium.com/@alexishevia/using-cors-in-express-cac7e29b005b) explains everything you need to do, and the official docs can be found [here](https://expressjs.com/en/resources/middleware/cors.html).
+Setting up CORS in Express is very easy, there's a middleware that does the work for us. The official docs can be found [here](https://expressjs.com/en/resources/middleware/cors.html#enabling-cors-pre-flight).
 
-For now, it is acceptable to just allow access from any origin. This makes development quite a bit easier but for any _real_ project, once you deploy to a production environment you will probably want to specifically block access from any origin _except_ your front-end website. The documentation above explains how to do this.
+For now, it is acceptable to just allow access from any origin. This makes development quite a bit easier but for any _real_ project, once you deploy to a production environment you will probably want to specifically block access from any origin _except_ your frontend website. The documentation above explains how to do this.
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
-1. [This article](https://medium.com/@shubhangirajagrawal/the-7-restful-routes-a8e84201f206) is a good resource for organizing RESTful APIs.
+1. [This article](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design) is a good resource for understanding and [organizing](https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/#h-use-nouns-instead-of-verbs-in-endpoint-paths) RESTful APIs.
```

Comment:
Subbed out this link - the original Medium article didn't contain much substance about REST design itself. I think it's best attribute was having a useful chart which I included in the curriculum directly.

The stackoverflow blogpots is more comprehensive although it verges on being daunting to read. I linked to the specific section on organizing APIs to try to maintain the same substance as before in case people didn't want to read the entire post.